src/components/icons/CalendarIcon.vue

```diff
@@ -0,0 +1,89 @@
+<template>
+  <svg
+    class="icon"
+    width="32"
+    height="32"
+    viewBox="0 0 32 32"
+    fill="none"
+    xmlns="http://www.w3.org/2000/svg"
+    @mouseenter="toggleHover"
+    @mouseleave="toggleHover"
+  >
+    <path
+      d="M25.75 7.75H6.25C5.42157 7.75 4.75 8.42157 4.75 9.25V25.75C4.75 26.5784 5.42157 27.25 6.25 27.25H25.75C26.5784 27.25 27.25 26.5784 27.25 25.75V9.25C27.25 8.42157 26.5784 7.75 25.75 7.75Z"
+      :stroke="iconColor"
+      stroke-width="1.5"
+      stroke-linecap="round"
+      stroke-linejoin="round"
+    />
+    <path
+      d="M10.75 10V4.75"
+      :stroke="iconColor"
+      stroke-width="1.5"
+      stroke-linecap="round"
+      stroke-linejoin="round"
+    />
+    <path
+      d="M21.25 10V4.75"
+      :stroke="iconColor"
+      stroke-width="1.5"
+      stroke-linecap="round"
+      stroke-linejoin="round"
+    />
+    <path
+      d="M13.75 18.25H9.25V22.75H13.75V18.25Z"
+      :stroke="iconColor"
+      stroke-width="1.5"
+      stroke-linecap="round"
+      stroke-linejoin="round"
+    />
+    <path
+      d="M18.25 13.75V22.75"
+      :stroke="iconColor"
+      stroke-width="1.5"
+      stroke-linecap="round"
+      stroke-linejoin="round"
+    />
+    <path
+      d="M22.75 13.75H13.75V22.75H22.75V13.75Z"
+      :stroke="iconColor"
+      stroke-width="1.5"
+      stroke-linecap="round"
+      stroke-linejoin="round"
+    />
+    <path
+      d="M13.75 18.25H22.75"
+      :stroke="iconColor"
+      stroke-width="1.5"
+      stroke-linecap="round"
+      stroke-linejoin="round"
+    />
+  </svg>
+</template>
+<script>
+import { ref, computed } from "vue";
+export default {
+  props: { color: { type: String, default: "#343541" } },
+  setup(props) {
+    const isHovered = ref(false);
+    function toggleHover() {
```

Comment:
I realized that when working on another branch. I went even further, I defined styles for hover on icon from button component using :deep(), since I only want the icon to change its color on hover only when it's inside a button.

Reply:
got it 🫡 
And I think there is another and probably the simplest solution for it 🤔 you can use the newest Vue magic, use `v-bind(VALUE)` directly in `style` block. you can get props with colors and handle these colors with css using `:hover` pseudoclass 👌 what you think about it? more info -> https://vuejs.org/api/sfc-css-features.html#v-bind-in-css

Reply:
I've read about it, it was tempting to use it in this case.
why I didn't?
Let's think about it. When do we want the icon to change its color? When hover on icon only? Or when the button, in which the icon is, is hovered? So is it actually a good idea to define :hover on icon?
This is how I tried to solve the problem: 
`      .custom-button:hover {
        background: #212229;
        color: white;
        & :deep(.icon *) {
          stroke: white;
        }
      }`

Reply:
totally agree 👍 and this way you have access to SCSS variables for the icon